src/components/editor/InputLookup.jsx

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 Stanford University see Apache2.txt for license
+import React, {Component} from 'react'
+import {asyncContainer, Typeahead} from 'react-bootstrap-typeahead'
+import PropTypes from 'prop-types'
+
+const AsyncTypeahead = asyncContainer(Typeahead)
+
+class InputLookup extends Component {
```

Comment:
so ... fundamentally the difference between this and the InputTarget component is that the InputTarget component does the typeahead rendering?   Because it's not really about the property type of "target" or "lookup" - correct?  It's about whether a certain valueConstraint is present and what is in it (saying how to look it up in QA - when it is a vocabulary vs when it isn't)?

Reply:
I think that we will need to actually use the property types "lookup" and "resource" in order to distinguish whether the fetched list is discrete or queryable. If it's "lookup" it's queryable (e.g. LCNAMES). If it's "resource" then we would look for presence of valueTemplateRefs to display buttons or not. I think the two are mutually exclusive in practice, but the presence of vTRs would override rendering the input field.

Reply:
Yes, ok to merge this and do follow on, but it now needs rebase.  ugh. sorry.